crates/services/gas_price_service/src/v1/uninitialized_task.rs

```diff
@@ -151,58 +168,62 @@ where
         // TODO: Add to config
         // https://github.com/FuelLabs/fuel-core/issues/2140
         let poll_interval = None;
+        if let Some(sequence_number) = self
+            .gas_price_db
+            .get_sequence_number(&metadata_height.into())?
+        {
+            self.da_source.set_last_value(sequence_number).await?;
+        }
```

Comment:
I don't understand how this part should work. You only set the sequencer number when you receive the bundle from the DA. 

<img width="659" alt="image" src="https://github.com/user-attachments/assets/53e60d16-448e-477d-a9f1-bff69b371352" />

Let's say you received the bundle with sequencer number `10` at L2 block height X. After that, you processed the L2 block at height X + 1. After that, the node was restarted.

During start up the `metadata_height` will be `X + 1`, but the `get_sequence_number` will return `None` in this case. So you will not continue from the sequencer number `10`.

Reply:
Ah. I see. So we need to set the most recent value again if it hasn't changed.